### PR TITLE
better screen reader support

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Some packages had elements which caused issues with screen readers. This PR addresses these by adding relevant aria tags.

### TODO:
~- [ ] icons are still rendered as svg without aria labels~
- [ ] correctly apply aria tags via styled components
- [ ] `transitFareWrapper` needs to be an aria button only when it has children.
- [ ] `transitFareWrapper` needs to be adjusted to match html spec
- [ ] internal `core-utils` versions need to be updated

### CODE REVIEW

File: packages/trip-form/src/SubmodeSelector/index.tsx [hunk 1]
```diff
@@ -57,9 +57,10 @@ export default function SubmodeSelector({
       {label && <LabelType>{label}</LabelType>}
       <RowType>
         {modes &&
-          modes.map(option => (
+          modes.map(( option, index ) => (
             <ModeButton
               key={option.id}
+              uniqueId={option.id + index}
```
Comment (binh-dam-ibigroup) [inline packages/trip-form/src/SubmodeSelector/index.tsx:63]: Sort props

File: packages/trip-form/src/ModeSelector/index.tsx [hunk 2]
```diff
@@ -53,6 +53,7 @@ export default function ModeSelector({
   const makeButton = (option: ModeSelectorOption): ReactElement => (
     <ModeButton
       key={option.id}
+      uniqueId={option.id}
```
Comment (binh-dam-ibigroup) [inline packages/trip-form/src/ModeSelector/index.tsx:56]: Sort props

File: packages/trip-form/src/ModeButton/index.tsx [hunk 3]
```diff
@@ -53,25 +57,31 @@ export default function ModeButton({
...
   return (
     <S.ModeButton className={className} style={style}>
       <S.ModeButton.Button
         aria-pressed={selected}
+        aria-label={title}
         className={`${activeClassName} ${disabledClassName}`}
         disabled={!enabled}
         onClick={onClick}
         title={title}
+        id={buttonId}
```
Comment (binh-dam-ibigroup) [inline packages/trip-form/src/ModeButton/index.tsx:77]: Sort props

File: packages/trip-form/src/ModeButton/index.tsx [hunk 4]
```diff
@@ -53,25 +57,31 @@ export default function ModeButton({
...
   const activeClassName = selected ? "active" : "";
   const disabledClassName = enabled ? "" : "disabled";
 
+  const buttonId = `${title?.replace(/ /g, "-")}-button-${uniqueId}`;
+
   return (
     <S.ModeButton className={className} style={style}>
       <S.ModeButton.Button
         aria-pressed={selected}
+        aria-label={title}
```
Comment (binh-dam-ibigroup) [inline packages/trip-form/src/ModeButton/index.tsx:72]: Sort props

File: packages/trip-form/src/DateTimeSelector/index.tsx [hunk 5]
```diff
@@ -220,6 +220,7 @@ export default function DateTimeSelector({
         {departureOptions.map(opt => (
           <ModeButton
             key={opt.type}
+            uniqueId={opt.type}
```
Comment (binh-dam-ibigroup) [inline packages/trip-form/src/DateTimeSelector/index.tsx:223]: Sort props

File: packages/trip-details/src/index.tsx [hunk 6]
```diff
@@ -181,7 +181,10 @@ export function TripDetails({
     fare = (
       <S.Fare>
         <TransitFareWrapper>
-          <summary style={{ display: fareKeys.length > 1 ? "list-item" : "" }}>
+          <summary
+            role="button"
```
Comment (binh-dam-ibigroup) [inline packages/trip-details/src/index.tsx:185]: Add an `aria-expanded` attribute. 

I would disagree with `role="button"` for the `<summary>` element, so maybe we shouldn't use `<summary>`? Also, in another PR, we'll need to implement `aria-controls` or `aria-owns` for this expandable component.

File: packages/itinerary-body/src/AccessLegBody/index.tsx [hunk 7]
```diff
@@ -104,6 +104,7 @@ class AccessLegBody extends Component<Props, State> {
             showLegIcon={showLegIcon}
           />
           <S.StepsHeader
+            role="button"
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/src/AccessLegBody/index.tsx:107]: Is this needed? It is already a `<button>`.

Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/index.tsx:107]: Do we need an `aria-controls` or `aria-owns` attribute?

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving subject to sort props and adding an `aria-expanded` attribute where noted.

File: packages/itinerary-body/src/AccessLegBody/index.tsx (hunk 7 above)
Comment (miles-grant-ibigroup) [reply packages/itinerary-body/src/AccessLegBody/index.tsx:107]: Yes the `role="button"` seems to not be needed, but `aria-expanded` is what seems to be needed.

File: packages/trip-details/src/index.tsx (hunk 6 above)
Comment (miles-grant-ibigroup) [reply packages/trip-details/src/index.tsx:185]: I followed https://codepen.io/scottohara/pen/aaJXYG. I think this is very much a summary but also think that there may be a better way than a random codepen to make it properly accessible

Comment (binh-dam-ibigroup) [reply packages/trip-details/src/index.tsx:185]: The component that reacts to clicking should be the button element... but I'm not sure which one it is!

Comment (binh-dam-ibigroup) [reply packages/trip-details/src/index.tsx:185]: And we'll have to fix this. Basically, the `<summary>` should be outside of `<details>`.
image

**Pull request closed without merging**